### DISCUSSION TITLE
Getting error while running foundry test

### DISCUSSION THREAD

**Discussion by skpkss** [Cyfrin/security-and-auditing-full-course-s23#217] [Q&A] [answered]
Hey,I'm currently in section 1: 3. Fuzzing and invariants.  When i am running basic test over MyContract ,it should pass ideally, but it is showing me EvmError: Revert. 
![cu1](https://github.com/user-attachments/assets/6b6dcc13-9988-4513-a1fb-915826df3023)
![cu2](https://github.com/user-attachments/assets/552291f8-0c95-4ae6-ac38-d1574a00fac1)

**Comment by EngrPips** [comment]
Hello @skpkss, Please run your test with the `-vvvv` flag and show us the test output.

**Reply by skpkss** [reply]
![cu3](https://github.com/user-attachments/assets/ec0a9efd-2a36-4c39-90fb-6739ec6a7649)

**Comment by EngrPips** [comment]
> Hey,I'm currently in section 1: 3. Fuzzing and invariants. When i am running basic test over MyContract ,it should pass ideally, but it is showing me EvmError: Revert. ![cu1](https://private-user-images.githubusercontent.com/107455544/354107675-6b6dcc13-9988-4513-a1fb-915826df3023.JPG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI1MDA5NzQsIm5iZiI6MTcyMjUwMDY3NCwicGF0aCI6Ii8xMDc0NTU1NDQvMzU0MTA3Njc1LTZiNmRjYzEzLTk5ODgtNDUxMy1hMWZiLTkxNTgyNmRmMzAyMy5KUEc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODAxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgwMVQwODI0MzRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lMmI5ZDQ2MWU3N2JmOTU4ZDZjNGZiZDA3MzM4Nzc0Y2RiYmZmMWY2NmJkMGY5ZGI3YjNkMzk3OWNjMzBmNTJkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.wouSzvBjdf98li9044NM7x41OmKmI6tOT-YaLlDtlgA) ![cu2](https://private-user-images.githubusercontent.com/107455544/354107690-552291f8-0c95-4ae6-ac38-d1574a00fac1.JPG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI1MDA5NzQsIm5iZiI6MTcyMjUwMDY3NCwicGF0aCI6Ii8xMDc0NTU1NDQvMzU0MTA3NjkwLTU1MjI5MWY4LTBjOTUtNGFlNi1hYzM4LWQxNTc0YTAwZmFjMS5KUEc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODAxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgwMVQwODI0MzRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wODhmMGMxODhiY2JkYmVkOGU2Mjg5MGU4NDI5ODdkYjU3YzliOGExYTc4MDdkODZkYjNkZDZkOTE3NDkxZWJmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.cUalR7ex2K3kOWjszKv8dvgS25x1ehDdWEkpJEFLpFc)

Why didn't you import `Test` into your `MyContractTest` file and make your test inherit from `Test` 

```js
import {Test} from "forge-std/Test.sol";
contract MyContractTest is Test {
...
}
```

**Reply by skpkss** [reply]
earlier i have tried this. still getting the same issue.
![cu4](https://github.com/user-attachments/assets/b80054c6-c292-4237-aa5d-dcb2e47b694f)

**Reply by EngrPips** [reply, marked as answer]
Hey @skpkss, why dont you have a `setUp` function inside of the this test contract?

Your test is suppose to look like the below.
```js
contract MyContractTest is Test {
MyContract exampleContract 

function setUp() public {
exampleContract = new MyContract();
}

function testIsAlwaysGetZero() public {
uint256 data = 0;
exampleContract.doStuff(data)
assert(exampleContract.shouldAlwaysBeZero == 0);
}
}
```

**Reply by skpkss** [reply]
Thanks.I'm a noob.

**Reply by EngrPips** [reply]
I am still a noob, too. Keep crushing it!

**Reply by skpkss** [reply]
Hey , when i my trying to install open zeppelin,it is giving submodule error.Any solution for that?
![cu5](https://github.com/user-attachments/assets/f4f38858-3596-4996-89e3-62b2585e6eee)

**Reply by EngrPips** [reply]
Please open your `lib` folder and check that you don't already have the `0penzeppelin` library in there already

**Reply by skpkss** [reply]
![cu6](https://github.com/user-attachments/assets/c6dd4523-0270-4c47-b1ca-7e71c431376c)
I have openzeppelin library. still getting this issue.

**Reply by EngrPips** [reply]
If you already have the `Openzepepelin` Library, why are you trying to install it again?

**Reply by skpkss** [reply]
because it's not able to import ERC20 from openzepplein

**Reply by EngrPips** [reply]
Interesting, If you are on Mac, use `command + p` a search bar will appear in `vscode`. Type `ERC20` in the search bar and see if it will find the file and locate where it is. You should use `cltr + p` on Windows and do the same things.

**Reply by skpkss** [reply]
I have erc20.sol but getting this issue.
![cu7](https://github.com/user-attachments/assets/1eae6a44-6d7f-4112-b11e-e0a7646bf4f5)

**Reply by EngrPips** [reply]
Try compiling your code and see if it compiles successfully. If it compiles successfully, you can ignore the
